### ISSUE TITLE
check_nrpe v3 and NSCP 0.5.0.62

### ISSUE THREAD

**Issue by kimllee** [mickem/nscp#440]
Hello,

Does any one succeded to use packets v3 with check_nrpe v3 (Centos 6.8) and NSCP 0.5.0.62 (on Windows 2012R2) ?

Thank you.

**Comment by garnier-quentin** [comment]
Unfortunately, NSCP is not compatible with protocol v3.

**Comment by box293** [comment]
This KB article should answer your question:

https://support.nagios.com/kb/article/nrpe-packet-size-explained.html

**Comment by box293** [comment]
@garnier-quentin check the KB article I just linked, it shows you how you can do it.

**Comment by garnier-quentin** [comment]
Yes you can use NRPE v3 with NSClient. But you cannot use the NRPE "dynamic" length payload. Sorry. It was not enough clear.

**Comment by box293** [comment]
Yeah it would be good if NSClient++ supported the dynamic length.

**Comment by kimllee** [comment]
Thank you for your quick answers guys. I already read the Nagios KB before posting a request here.
Whether the length payload is  "static set" or dynamic I have to use the nrpe v2 packets to avoid 
"Remote x.x.x.x does not support Version 3 Packets " message.
I guess I have to wait until NSCP is fully compatible with NRPEv3.

**Comment by mickem** [comment]
I'd happily accept pull requests for an updated NRPE plugin with support for version 3

**Comment by dnsmichi** [comment]
Guess this should be added to the troubleshooting docs meanwhile. Debian Stretch provides one with that, and so does EPEL. I'll see what I can do.

**Comment by box293** [comment]
Some great discussion points here, I've create a KB article to explain how to resolve the issue. Either use v2 packets OR create a rsyslog rule to discard the message.

https://support.nagios.com/kb/article/nrpe-remote-does-not-support-version-3-packets-786.html

@kimllee @dnsmichi does this help?
